terraform/network.tf

```diff
@@ -0,0 +1,176 @@
+  tags = {
+    Tier = "private",
+    Name = "private subnet ${element(var.availability_zones, count.index)}"
+  }
+}
+
+resource "aws_subnet" "public" {
+  vpc_id                  = aws_vpc.main.id
+  cidr_block              = element(var.public_subnets, count.index)
+  availability_zone       = element(var.availability_zones, count.index)
+  count                   = length(var.public_subnets)
+  map_public_ip_on_launch = true
+  tags = {
+    Tier = "public"
+    Name = "public subnet ${element(var.availability_zones, count.index)}"
+  }
+}
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.main.id
+}
+
+resource "aws_route" "public" {
+  route_table_id         = aws_route_table.public.id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.main.id
+}
+
+resource "aws_route_table_association" "public" {
+  count          = length(var.public_subnets)
+  subnet_id      = element(aws_subnet.public.*.id, count.index)
+  route_table_id = aws_route_table.public.id
+}
+resource "aws_nat_gateway" "main" {
+  count         = length(var.public_subnets)
+  allocation_id = element(aws_eip.nat.*.id, count.index)
+  subnet_id     = element(aws_subnet.public.*.id, count.index)
+  depends_on    = [aws_internet_gateway.main]
+}
+
+resource "aws_eip" "nat" {
+  count = length(var.public_subnets)
+  vpc   = true
+}
+resource "aws_route_table" "private" {
+  count  = length(var.private_subnets)
+  vpc_id = aws_vpc.main.id
+}
+
+resource "aws_route" "private" {
+  count                  = length(compact(var.private_subnets))
+  route_table_id         = element(aws_route_table.private.*.id, count.index)
+  destination_cidr_block = "0.0.0.0/0"
+  nat_gateway_id         = element(aws_nat_gateway.main.*.id, count.index)
+}
+
+resource "aws_route_table_association" "private" {
+  count          = length(var.private_subnets)
+  subnet_id      = element(aws_subnet.private.*.id, count.index)
+  route_table_id = element(aws_route_table.private.*.id, count.index)
+}
+
+resource "aws_security_group" "alb" {
+  name   = "${var.name}-sg-alb-${var.environment}"
+  vpc_id = aws_vpc.main.id
+
+  ingress {
+    protocol         = "tcp"
+    from_port        = 80
+    to_port          = 80
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    protocol         = "tcp"
+    from_port        = 443
+    to_port          = 443
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    protocol         = "-1"
+    from_port        = 0
+    to_port          = 0
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+}
+
+resource "aws_security_group" "ecs_tasks" {
+  name   = "${var.name}-sg-task-${var.environment}"
+  vpc_id = aws_vpc.main.id
+
+  ingress {
+    protocol         = "tcp"
+    from_port        = var.container_port
+    to_port          = var.container_port
+    cidr_blocks      = ["0.0.0.0/0"]
```

Comment:
Ideally, this would be restricted to just the security group of the load balancer.

Reply:
Not sure I understand

Reply:
The only thing that makes a connection to the ecs tasks is the load balancer, right?

If that's the case you can restrict the ecs security group to only allow connections from the load balancer security group using the `secuirty_group` option for ingress/egress blocks.  

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group#security_groups

Reply:
so could I also use that for the egress block of the alb as well??

Reply:
That was a dumb question... I was in a different headspace :)